File: app/src/main/java/com/hxh/kt10/java/PlainOldSingleton.java
package com.hxh.kt10.java;

/**
 * Created by HXH on 2017/6/1 0001.
 * 普通的单例
 */

public class PlainOldSingleton {

    private static PlainOldSingleton INSTANCE = new PlainOldSingleton();

    private PlainOldSingleton() {
        System.out.println("PlainOldSingleton");
    }

    public static PlainOldSingleton getInstance() {
        return INSTANCE;
    }
}

File: app/src/main/java/com/hxh/kt10/kotlin/PlainOldSingleton.kt
package com.hxh.kt10.kotlin

/**
 * Created by HXH on 2017/6/1 0001.
 *
 */
object PlainOldSingleton {
    fun sayHello() {
        println("hello")
    }
}

fun main(args: Array<String>) {
    val s: PlainOldSingleton = PlainOldSingleton;
    s.sayHello()
}


